pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +225,40 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
```

Comment:
I'm not quite convinced.You have to iterate over list of Pods and get logs for each of them. You're basically doing just that in here. What I'm asking is to move that code into the other file (`pkg/kubectl/cmd/util/factory.go`) which actually invokes client calls to get logs for each container. If that is not clear lemme know, I can submit a PR against your branch with the solution.